packages/asc-ui/src/components/Radio/RadioStyle.ts

```diff
@@ -1,29 +1,139 @@
-import styled from '@datapunt/asc-core'
+import styled, { css } from '@datapunt/asc-core'
+import { themeColor, themeSpacing } from '../../utils'
+import { outlineStyle } from '../../utils/themeUtils'
 
-interface Props {
+enum Variants {
+  primary,
+  secondary,
+  tertiary,
+}
+
+export type Props = {
+  variant?: keyof typeof Variants
+  disabled?: boolean
+  error?: boolean
+  name?: string
   id: string
-  name: string
-  value: string
-  defaultChecked: boolean
-  disabled: boolean
+  defaultChecked?: boolean
+  onChange?: Function
+}
+
+const getVariantColor = () => ({
+  variant,
+}: {
+  variant?: keyof typeof Variants
+}) => {
+  switch (variant) {
+    case 'primary':
+      return css`
+        ${themeColor('primary', 'main')};
+      `
+
+    case 'secondary':
+      return css`
+        ${themeColor('secondary', 'main')};
+      `
+
+    default:
+      return css`
+        ${themeColor('tint', 'level7')};
+      `
+  }
 }
 
-const RadioStyle = styled.input.attrs<Props>({
+const RadioStyle = styled.input.attrs({
   type: 'radio',
-  id: ({ id }: Props) => id,
-  name: ({ name }: Props) => name,
-  value: ({ value }: Props) => value,
-  defaultChecked: ({ defaultChecked }: Props) => defaultChecked,
-  disabled: ({ disabled }: Props) => disabled,
-})`
-  cursor: pointer;
-  height: 28px;
+})<Props>`
+  top: 0;
   left: 0;
+  width: 100%;
+  cursor: inherit;
+  height: 100%;
   margin: 0;
-  opacity: 0;
+  padding: 0;
   position: absolute;
-  top: 0;
-  width: 28px;
-  zoom: 1;
+  opacity: 0;
 `
+
 export default RadioStyle
+
+type StyleOnlyProps = {
+  variant?: keyof typeof Variants
+  checked?: boolean
+  disabled?: boolean
+  error?: boolean
+  focus?: boolean
+}
+
+const RadioCircleStyle = styled.span<StyleOnlyProps>`
+  width: ${themeSpacing(6)};
+  height: ${themeSpacing(6)};
+  color: ${themeColor('tint', 'level5')};
+  border: 1px solid;
+  border-radius: 50%;
+  ${({ checked }) =>
+    checked &&
+    css`
+      color: ${getVariantColor()};
+      &::after {
+        content: '';
+        position: absolute;
+        height: 16px;
```

Comment:
themeSpacing?

Reply:
No `themeSpacing` on `height` and `width`.